### PR TITLE
Add cmake build support for windows, linux, macos

### PR DESCRIPTION
**Pull request by kippesp**
- checked with clang 13,14,15, gcc 11, MSVS 2019,2022

### CODE REVIEW

File: FloatNumber.h [hunk 1]
```diff
@@ -26,8 +26,10 @@ template <
 struct FloatDefinition
 {
     // The warning is bogus, since the shift result is not actually used in such a case.
+#if defined(_WIN32) || defined(_WIN64)
```
Comment (fdwr) [inline FloatNumber.h]: Given this test is more about which compiler than OS, would `_MSC_VER` be more appropriate?

File: BiNums.cpp [hunk 2]
```diff
@@ -253,8 +253,8 @@ enum class NumericPrintingFlags : uint32_t
 inline constexpr NumericPrintingFlags operator ~ (NumericPrintingFlags lhs) { return NumericPrintingFlags(~uint32_t(lhs)); }
 inline constexpr NumericPrintingFlags operator & (NumericPrintingFlags lhs, NumericPrintingFlags rhs) { return NumericPrintingFlags(uint32_t(lhs) & uint32_t(rhs)); }
 inline constexpr NumericPrintingFlags operator | (NumericPrintingFlags lhs, NumericPrintingFlags rhs) { return NumericPrintingFlags(uint32_t(lhs) & uint32_t(rhs)); }
-inline constexpr bool operator == (NumericPrintingFlags lhs, uint32_t rhs) { return uint32_t(lhs) == 0; }
-inline constexpr bool operator != (NumericPrintingFlags lhs, uint32_t rhs) { return uint32_t(lhs) != 0; }
+inline constexpr bool operator == (NumericPrintingFlags lhs, uint32_t /* rhs */) { return uint32_t(lhs) == 0; }
```
Comment (fdwr) [inline BiNums.cpp:256]: @fdwr (note to self) Fix these two lines and PrintAllPrintingFormats afterward.

File: FloatNumber.h (hunk 1 above)
Comment (kippesp) [reply FloatNumber.h]: Got it.  Made this change to two other areas--same situation.

**Review by fdwr — APPROVED** [review]
Thanks Paul.

**Comment by fdwr** [comment]
Btw, how did you test so many versions? Did you have that many distinct versions of clang installed, or use some remote build service?

**Pull request merged**